pkgs/by-name/fr/freecad/package.nix

```diff
@@ -210,6 +215,14 @@ stdenv.mkDerivation (finalAttrs: {
       mechanical engineering and architecture. Whether you are a hobbyist, a
       programmer, an experienced CAD user, a student or a teacher, you will feel
       right at home with FreeCAD.
+
```

Comment:
This whole string should be in a README.md file, instead of cluttering the stdout.

Reply:
Thanks for educating me on the problem about the configuration via input parameters. But, to be honest, I don't understand what your suggestion is. How should I implement optional ifcopenshell support? Is my code ok or is there a better way to do it?


>>   I think all packages should just do that
>
> All packages can't do that. Indeed no package can do that.

Here I'm speaking specifically about documenting in longDescription so that it appears in search.nixos.org (I feel like we are mixing this with the `ifcSupport` option maybe?). Are you saying that there is no way to document the different options of a package there and that people do need to check the source code to learn about the customization possibilities?

Reply:
> so that it appears in search.nixos.org

It looks like a deviation of purpose of `longDescription`.

This field describes the software, not the input parameters of the build script.
Further, it is not "a field to be shown in search.nixos.org engine".

Further, the [discoverability of Nixpkgs is very problematic](https://github.com/NixOS/nixpkgs/issues/222431), and such workaround is far from helpful.

Reply:
I feel like we are mixing two orthogonal topics here : the issue you linked is about nixos options installing packages under the hood. Here I'm talking about package options that changes the derivation itself. These options are available for nix/nixpkgs itself, not only for nixos. Both are kinda related yes, but not entirely.

I agree `longDescription` was probably not meant for that, but in my opinion it's still better than nothing. I'd argue to keep this commit , but as you are set on removing this, I'll amend it and but this in a README.